vortex-array/src/stats/stats_set.rs

```diff
@@ -107,6 +95,44 @@ impl StatsSet {
     pub fn iter(&self) -> impl Iterator<Item = &(Stat, Precision<ScalarValue>)> {
         self.values.iter()
     }
+
+    /// Get value for a given stat
+    pub fn get(&self, stat: Stat) -> Option<Precision<ScalarValue>> {
+        self.values
+            .iter()
+            .find(|(s, _)| *s == stat)
+            .map(|(_, v)| v.clone())
+    }
+
+    /// Length of the stats set
+    pub fn len(&self) -> usize {
+        self.values.len()
+    }
+
+    /// Check whether the statset is empty
+    pub fn is_empty(&self) -> bool {
+        self.values.is_empty()
+    }
+
+    /// Get scalar value of a given dtype
+    pub fn get_as<T: for<'a> TryFrom<&'a Scalar, Error = VortexError>>(
+        &self,
+        stat: Stat,
+        dtype: &DType,
+    ) -> Option<Precision<T>> {
+        self.get(stat).map(|v| {
+            v.map(|v| {
+                T::try_from(&Scalar::new(dtype.clone(), v)).unwrap_or_else(|err| {
+                    vortex_panic!(
```

Comment:
if the Scalar is somehow null or empty for the stat (as opposed to missing), we'll panic?

I think this is fine, because we don't allow Null scalars for Min, Max, etc... we just store NullCount instead. But just thinking out loud, should make sure we document this well / have good tests for it. (Separate from this PR)